src/ui/Island/Cursor/CursorLabel.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react"
+import Icon from "shared/components/Icon"
+import { ReflectCursor, ReflectUserInfo } from "shared/types"
+
+export type CursorLabelProps = {
+  cursor: ReflectCursor
+  userInfo: ReflectUserInfo
+  extraCursor: boolean
+}
+
+export default function CursorLabel({
+  cursor,
+  userInfo,
+  extraCursor,
+}: CursorLabelProps) {
+  if (!cursor) return null
```

Comment:
you don't need this additional check when you check it during mapping in src/ui/Island/IslandPresence.tsx and you also set it required in type.

Reply:
This actually should be of type `ReflectCursor | null`, my mistake

Reply:
Why we can't add this as a required prop?